src/vchan_cmds.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <domain.h>
+#include <vch.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/shell/shell.h>
+
+LOG_MODULE_REGISTER(vchan_cmd);
+
+#define ECHO_SERVER_THREAD_PRIO    7
+static struct k_thread echo_thread;
+static struct k_thread *pecho_thread = NULL;
+static struct vch_handle handle;
+
+static K_THREAD_STACK_DEFINE(main_stack, 4096);
+static volatile int vchan_server_stop = 0;
+#define PATH_TEMPLATE "/local/domain/%d/echo"
+char path[30];
+
+static void vchan_echo_server(void *p1, void *p2, void *p3)
+{
+	struct vch_handle *h = (struct vch_handle *)p1;
+	int size;
+	static char buf[CONFIG_VCHAN_ECHO_BUFFER_SIZE];
+
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+	LOG_DBG("Server started\n");
+	while (!vchan_server_stop) {
+		size = vch_read(h, buf, sizeof(buf));
+		if (size > 0) {
+			vch_write(h, buf, size);
+			continue;
+		} else {
```

Comment:
Just suggestion. you don't need else here because you're using continue. It's up to you but I think it will make code lool neat.

Reply:
Redesigned